fastFM/ffm2.pyx

```diff
@@ -77,11 +77,25 @@ def ffm_predict(np.ndarray[np.float64_t, ndim = 1] w_0,
 
     return y
 
+# The main piece of the glue between Python, Cython and C++
+# In essence it wraps the python function so it can be used in C++ space.
+# Convert the python function from a pointer back into a python object and invoke
+# with other parameters. (For now just one, `current_iter`)
+cdef void fit_callback_wrapper(int current_iter, void* python_function):
```

Comment:
My thoughts on this:
- callback is not None. If callback is None inform C++ side somehow so no cycles are wasted invoking the callback. This might be harder than it sounds.
- callback corresponds to `fit_callback_t`. One way to check is to catch `TypeError` which is raised if a function is invoked with less or more than the required number of arguments.
- passing all parameters as `dict` and invoking the python callback via **kwargs for greater flexibility